niceplots/utils.py

```diff
@@ -624,6 +625,167 @@ def plotColoredLine(
+
+    # Go through the colors and only take the color information (not transparency)
+    for i in range(len(colors)):
+        if colors[i][0] != "#":
+            raise ValueError("Colors specified as a string must start with a #")
+        colors[i] = colors[i][0:7]
+
+    # Sum categories and collapse subcategories
+    innerVals = []
+    innerLabels = []
+    outerVals = []
+    outerLabels = []
+    total = 0.0
+    maxSubcat = 0.0
+    for cat, val in data.items():  # top level categories
+        innerLabels.append(cat)
+        innerVals.append(0.0)
+
+        # Max number of subcategories for a given category
+        maxSubcat = max(maxSubcat, len(val))
+
+        for subcat, subcatVal in val.items():
+            innerVals[-1] += subcatVal
+            total += subcatVal
+
+            outerVals.append(subcatVal)
+            outerLabels.append(subcat)
+
+    # Define alphas if not specified
+    if alphas is None:
+        alphas = np.linspace(0.75, 0.95, maxSubcat)[-1::-1]
+
+    innerColors = [colors[i] for i in range(len(data))]
+    outerColors = []
+    iCat = 0
+    for catVals in data.values():
+        numSubcats = len(catVals)
+        for iSubcat in range(numSubcats):
+            outerColors.append(colors[iCat] + float.hex(alphas[iSubcat])[4:6])
+        iCat += 1
+
+    # Nested plot fitting params
+    size = 0.3
+    buffer = 0.01
+
+    # Create figure if it's not passed in
+    returnFig = False
+    if ax is None or fig is None:
+        fig, ax = plt.subplots()
+        returnFig = True
+
+    # Set keyword arguments
+    outerKwargDefaults = {
+        "radius": 1.0,
+        "colors": outerColors,
+        "wedgeprops": dict(width=size, edgecolor=None),
+        "textprops": dict(rotation_mode="anchor", va="center", ha="center", color="w"),
+        "labels": outerLabels,
+        "rotatelabels": False,
+        "labeldistance": 0.85,
+    }
+    innerKwargDefaults = {
+        "radius": 1.0 - size - buffer,
+        "colors": innerColors,
+        "wedgeprops": dict(width=size, edgecolor=None),
+        "textprops": dict(rotation_mode="anchor", va="center", ha="center", color="w"),
+        "labels": innerLabels,
+        "rotatelabels": False,
+        "labeldistance": 0.75,
+    }
+
+    # Update kwargs
+    for outerKey, outerKwargVal in outerKwargDefaults.items():
+        if outerKey not in outerKwargs:
+            outerKwargs[outerKey] = outerKwargVal
+    for innerKey, innerKwargVal in innerKwargDefaults.items():
+        if innerKey not in innerKwargs:
+            innerKwargs[innerKey] = innerKwargVal
+
+    # Create the pie charts
+    outerWedges, outerText = ax.pie(outerVals, **outerKwargs)
+    innerWedges, innerText = ax.pie(innerVals, **innerKwargs)
+
+    ax.set(aspect="equal")
+
+    if returnFig:
+        return innerWedges, innerText, outerWedges, outerText, fig, ax
+    else:
+        return innerWedges, innerText, outerWedges, outerText
```

Comment:
Not sure its useful or not, but would it make sense to place the wedges and text in a dict similar to the input `data` dict? This way the input and output to the function is similar in format. Currently, it matches the input to the `pie` function.

Reply:
I like the idea of having a parallel structure to the input dictionary so it's easier to find text and wedge data, I'll change it so it does that